### PR TITLE
feat: replace hardcoded server config with TOML configuration

### LINKED ISSUE #4: Should have a config file to specify server list, rather than hard-coding it
Then use a cmd-line option to specify the config file.

### PR DESCRIPTION
**Pull request by imprvhub**
- Remove hardcoded server configuration from main code
- Add server configuration via TOML files
- Add config file loading with fallback paths
- Create ServerConfig.from_config method for parsing
- Update project dependencies to include tomli

Closes #4

### CODE REVIEW

File: lsp-multiplexer.py [hunk 1]
```diff
@@ -506,12 +551,32 @@ def format(self, record):
...
+
+    # Fall back to default configuration if no config file found
+    if not server_configs:
+        logger.info("No config file found, using default configuration")
+        server_configs = [
+            ServerConfig(command=["pyright-langserver", "--stdio"]),
+            # Add other default servers here
+        ]
+
+    multiplexer = LSPMultiplexer(server_configs)
```
Comment (garyo) [inline lsp-multiplexer.py:579]: The only change I see is now you're initializing LSPMultiplexer with a list of ServerConfigs, so you should adjust the type signature of LSPMultiplexer.__init__ to reflect that. For instance:
```python
    def __init__(self, server_configs: List[ServerConfig]) -> None:
```
With that, `pyright` should pass again. (Hmm, I should set up a quick pyright test CI action.)

**Review by garyo — CHANGES_REQUESTED** [review]
Looks great to me. One minor change, otherwise ready to merge.

File: lsp-multiplexer.py (hunk 1 above)
Comment (imprvhub) [reply lsp-multiplexer.py:579]: Done! [7520b78](https://github.com/garyo/lsp-multiplexer/pull/5/commits/7520b7882ded0445d5b84a3cb2db2d932c69d74f)

**Review by garyo — APPROVED** [review]
Thanks!

**Pull request merged**